File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.java
class Solution {
    public int solution(String s) {
        int answer = 0;
        
        char firstC = s.charAt(0); // 첫번재 문자 
        
        int firstCnt=0;
        int nextCnt=0;
        for(int i=0; i<s.length(); i++) {
            char ch = s.charAt(i);
            // 첫번째 문자와 다음문자가 같을 경우
            if(firstC==ch) firstCnt++;
            // 첫번째 문자와 다음문자가 다를 경우
            else nextCnt++;
            //System.out.println(ch+" "+firstCnt+" "+nextCnt);
            // answer를 카운팅해주고 다음 문자 탐색
            if(firstCnt==nextCnt) {
                answer++;
                firstCnt=0;
                nextCnt=0;
                if(i+1<s.length()) firstC = s.charAt(i+1);
            
            } else if(i==s.length()-1) answer++;
            
        }
    
        return answer;
    }
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.js
function solution(s) {
    var answer = 0;
    let cnt = 0;
    let other = 0;
    let first = s[0];
    for(let i=0; i<s.length; i++) {
        if(cnt==other) {
            answer++;
            first=s[i];
        }
        
        if(first===s[i]) {
            cnt++;
        } else {
            other++;
        }

    }
    return answer;
}